src/sources/amqp.rs

```diff
@@ -236,6 +222,42 @@ fn populate_event(
         "offset",
         keys.delivery_tag,
     );
+
+    log_namespace.insert_vector_metadata(
+        log,
+        path!(log_schema().source_type_key()),
+        path!("source_type"),
+        Bytes::from_static(AmqpSourceConfig::NAME.as_bytes()),
+    );
+
+    // This handles the transition from the original timestamp logic. Originally the
+    // `timestamp_key` was populated by the `properties.timestamp()` time on the message, falling
+    // back to calling `now()`.
```

Comment:
I'm really glad we have a comment here explaining this!

Reply:
😅 it definitely helped to write it out to make sure I got the logic right as well.